apis/Google.Cloud.BigQuery.V2/Google.Cloud.BigQuery.V2/JobCreationOptions.cs

```diff
@@ -59,5 +62,33 @@ public abstract class JobCreationOptions
         /// Label keys must start with a letter.
         /// </summary>
         public IDictionary<string, string> Labels { get; set; }
+
+        /// <summary>
+        /// The ID of the session in which to create the job.
+        /// If specified, this is used to associates the job with a given session.
+        /// </summary>
+        public string SessionId { get; set; }
+
+        /// <summary>
+        /// If true, creates a new session, where the session ID will be generated by the serve.
+        /// If false, runs query with an existing session_id passed in ConnectionProperty,
+        /// otherwise runs the load job in non-session mode.
+        /// </summary>
+        public bool? CreateSession { get; set; }
+
+        /// <summary>
+        /// Creates a list of <see cref="ConnectionProperty"/> i.e ConnectionProperties.
+        /// Currently only SessionId is supported in connection property.
+        /// For more information refer to https://cloud.google.com/bigquery/docs/reference/rest/v2/ConnectionProperty.
+        /// </summary>
+        protected IList<ConnectionProperty> GetConnectionProperties()
```

Comment:
Let's make this private protected so we can change it without any breaking changes. (Sorry, I should have suggested that before.)

But something about the split between this method and checking for SessionId being null feels brittle. I think we should definitely chat.